### PR TITLE
Add UrbanDictionary content provider

### PR DESCRIPTION
**Pull request by WilliamBZA**
To address this issue: https://github.com/davidfowl/JabbR/issues/518

Standard approach, get parameters, make request to API and render result.

### CODE REVIEW

File: JabbR/ContentProviders/UrbanDictionaryContentProvider.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                return null;
+            }
+
+            var apiQuery = string.Format(_urbanDictionaryAPIURL,
+                string.IsNullOrWhiteSpace(term) ? string.Empty : string.Format("term={0}", term),
+                string.IsNullOrWhiteSpace(defid) ? string.Empty : string.Format("&defid={0}", defid));
+
+            return FetchArticle(new Uri(apiQuery)).Then(result =>
+                {
+                    var list = result.list;
```
Comment (samandmoore) [inline JabbR/ContentProviders/UrbanDictionaryContentProvider.cs:42]: Do you need to do a check here to verify that the result came back with data, or is that handled implicitly somewhere in the core code? I can't remember.

**Comment by samandmoore** [comment]
Do you think you could squash your commits before we merge this so we don't get that hideous merge commit message?

File: JabbR/ContentProviders/UrbanDictionaryContentProvider.cs (hunk 1 above)
Comment (WilliamBZA) [inline JabbR/ContentProviders/UrbanDictionaryContentProvider.cs:42]: Yeah, it breaks with not found items. Will fix and resubmit sometime.

File: JabbR/ContentProviders/UrbanDictionaryContentProvider.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            string term = parameters["term"];
+
+            if (string.IsNullOrWhiteSpace(term) && string.IsNullOrWhiteSpace(defid))
+            {
+                return null;
+            }
+
+            var apiQuery = string.Format(_urbanDictionaryAPIURL,
+                string.IsNullOrWhiteSpace(term) ? string.Empty : string.Format("term={0}", term),
+                string.IsNullOrWhiteSpace(defid) ? string.Empty : string.Format("&defid={0}", defid));
```
Comment (davidfowl) [inline JabbR/ContentProviders/UrbanDictionaryContentProvider.cs:38]: Big S for String. methods

Comment (WilliamBZA) [inline JabbR/ContentProviders/UrbanDictionaryContentProvider.cs:38]: Just the methods or the variables too? (private static readonly string _contentFormat = blah vs private static readonly String _contentFormat = blah)

Comment (davidfowl) [inline JabbR/ContentProviders/UrbanDictionaryContentProvider.cs:38]: Just the methods

**Comment by WilliamBZA** [comment]
Will resubmit when I get time again

**Pull request closed without merging**